test/comparators_test.go

```diff
@@ -0,0 +1,58 @@
+package test_test
+
+import (
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/LUSHDigital/core/test"
+	"github.com/google/go-cmp/cmp"
+)
+
+type TestError struct{ error }
+
+func TestErrorTypeComparator(t *testing.T) {
+	testErr := TestError{errors.New("oops")}
+	someErr := errors.New("ouch")
+	opts := cmp.Options{
+		test.ErrorTypeComparer,
+	}
+	if cmp.Equal(testErr, someErr, opts) {
+		t.Fatalf("expected %T and %T to be of different types\n", someErr, testErr)
+	}
+}
```

Comment:
I honestly think we should have more test cases for when the values are nil, etc.